test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -1355,10 +1355,15 @@ def test_ddp_zero_overlap(
         self.dist_init(self.rank, self.world_size, backend)
         hook_constructor = hook_with_zero_step if not use_interleaved_hook \
             else hook_with_zero_step_interleaved
-        self._test_ddp_zero_overlap(
-            device, hook_constructor, gradient_as_bucket_view, static_graph,
-            shard_buckets=shard_buckets,
-        )
+
+        # Disable DDP + ReplicatedTensor since ZeroRedundancyOptimizer
+        # modifies the model parameters in place.
+        from torch.nn.parallel._replicated_tensor_ddp_utils import _ddp_replicated_tensor
+        with _ddp_replicated_tensor(False):
```

Comment:
Curious if there are future features developed in DDP:

1. Will they be required / enforced to have replicated tensor support?
2. Will we eventually be in some sort of situation where there is a subset of features supported by replicated tensor and a subset that is not?
3. If a user is developing a feature such as one that may modify model params in place, will the developer quickly be able to realize the issue may be due to DDP running with replicated tensor and not native DDP itself?

Reply:
1. Ideally we should try to ensure we can make this work and only disable support if we can't find a way around.
2. Currently there are only 2, zero redundancy optimizer and the stateless API. I think zero redundancy can actually be fixed by rebuilding the replicated_module when param.data is modified. Hopefully we can figure out a way to fix stateless too.
3. This might be tricky and not obvious to the developer why things might be failing here.